packages/webpack-plugin/src/runtime-inject.ts

```diff
@@ -0,0 +1,195 @@
+        }
+
+        replacePlaceholderExport(source, `{__classes__:true}`, stylableBuildMeta.exports.classes);
+        replacePlaceholderExport(source, `{__namespace__:true}`, stylableBuildMeta.namespace);
+
+        const usedExports = moduleGraph.getUsedExports(module, runtime);
+        if (typeof usedExports === 'boolean') {
+            if (usedExports) {
+                runtimeRequirements.add(StylableRuntimeStylesheet.name);
+            }
+        } else if (!usedExports) {
+            runtimeRequirements.add(StylableRuntimeStylesheet.name);
+        } else if (
+            usedExports.has('st') ||
+            usedExports.has('style') ||
+            usedExports.has('cssStates')
+        ) {
+            runtimeRequirements.add(StylableRuntimeStylesheet.name);
+        }
+    }
+}
+
+export class StylableRuntimeStylesheet extends RuntimeModule {
+    constructor() {
+        super('stylable stylesheet', 10);
+    }
+    generate() {
+        return `(${stylesheet})(__webpack_require__)`;
+    }
+}
+
+export class StylableRuntimeInject extends RuntimeModule {
+    constructor() {
+        super('stylable inject', 10);
+    }
+    generate() {
+        return `(${injectStyles})(__webpack_require__)`;
+    }
+}
+
+export class LoadCSS extends RuntimeModule {
+    constructor() {
+        super('stylable load css', 10);
+    }
+    generate() {
+        return `${RuntimeGlobals.ensureChunkHandlers}.stcss = ()=>{/*load css for chunk*/}`;
+    }
+}
+
+function replacePlaceholderExport(
+    source: sources.ReplaceSource,
+    replacementPoint: string,
+    value: string | Record<string, string>
+) {
+    const i = source.source().indexOf(replacementPoint);
+    if (!i) {
+        throw new Error(`missing ${replacementPoint} from stylable loader source`);
+    }
+    source.replace(
+        i,
+        i + replacementPoint.length - 1,
+        JSON.stringify(value),
+        `${replacementPoint} optimizations`
+    );
+}
+
+export function injectRuntimeModules(name: string, compilation: Compilation) {
+    compilation.hooks.runtimeRequirementInModule
+        .for(StylableRuntimeInject.name)
+        .tap(name, (_module, set) => {
+            set.add(RuntimeGlobals.require);
+        });
+
+    compilation.hooks.runtimeRequirementInModule
+        .for(StylableRuntimeStylesheet.name)
+        .tap(name, (_module, set) => {
+            set.add(RuntimeGlobals.require);
+        });
+
+    compilation.hooks.runtimeRequirementInTree
+        .for(StylableRuntimeInject.name)
+        .tap(name, (chunk, _set) => {
+            compilation.addRuntimeModule(chunk, new StylableRuntimeInject());
+        });
+
+    compilation.hooks.runtimeRequirementInTree
+        .for(StylableRuntimeStylesheet.name)
+        .tap(name, (chunk, _set) => {
+            compilation.addRuntimeModule(chunk, new StylableRuntimeStylesheet());
+        });
+}
+
+makeSerializable(StylableRuntimeDependency, __filename, StylableRuntimeDependency.name);
```

Comment:
what's this? why does it include __filename? move it next to class

Reply:
when webpack uses file-system cache it stores when to recall this type from. as I'm writing it I'm thinking maybe it supports requests like '@stylable/webpack-plugin' and named export key. need to check this.